### ISSUE TITLE
📝 Unable to format `package.json`

### ISSUE THREAD

**Issue by beeb** [biomejs/biome#1668]
### Environment information

```bash
CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.19.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.3"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Configuration

```JSON
{
  "$schema": "https://biomejs.dev/schemas/1.5.3/schema.json",
  "formatter": {
    "enabled": true,
    "indentStyle": "space",
    "lineWidth": 120
  },
  "javascript": {
    "formatter": {
      "quoteStyle": "single",
      "semicolons": "always",
      "jsxQuoteStyle": "single"
    }
  },
  "linter": {
    "enabled": true
  },
  "files": {
    "ignore": [
      "node_modules/*",
      ".env",
      ".env.*",
      "package-lock.json"
    ]
  }
}
```

### Playground link

https://codesandbox.io/p/devbox/runtime-platform-kkfs9k?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clrtbstcy0005286dygkqnt0k%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clrtbstcy0002286dgx32nd1z%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clrtbstcy0003286db51cb1st%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clrtbstcy0004286dwlkcpgek%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clrtbstcy0002286dgx32nd1z%2522%253A%257B%2522id%2522%253A%2522clrtbstcy0002286dgx32nd1z%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrtbstcy0001286d298idia9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.codesandbox%252FDockerfile%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clrtbstcy0001286d298idia9%2522%257D%252C%2522clrtbstcy0004286dwlkcpgek%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clrtbstcy0004286dwlkcpgek%2522%257D%252C%2522clrtbstcy0003286db51cb1st%2522%253A%257B%2522id%2522%253A%2522clrtbstcy0003286db51cb1st%2522%252C%2522activeTabId%2522%253A%2522clrtbz4ej0064286cximut6xu%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrtbz4ej0064286cximut6xu%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clrtbz4fo002eefip81iwe7pk%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

### Reproduce

```
npx @biomejs/biome format --write ./package.json
```

Result:

```
➜  /workspace git:(master) ✗ npx @biomejs/biome format --write ./package.json
Formatted 0 file(s) in 123µs
internalError/io ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ No files were processed in the specified paths.
  
```

### Comment

I know about https://biomejs.dev/guides/how-biome-works/#protected-files but here we are not linting, simply formatting. I didn't think biome would prevent formatting of the package.json, and `prettier` has no problem doing it. The page above doesn't indicate any justification for ignoring those files either.

What's more, the VSCode extension is happy to process the file when `Format on save` is enabled, so it seems the capability to format this file is present.

This is a regression that appeared first in 1.4.1. Prior versions are happy to process the command above

**Comment by Conaclos** [comment]
I share your point. I am not sure how we should solve this issue. Maybe by introducing a configuration `files.ignoreProtectued` (turned on by default)?

**Comment by cdaringe** [comment]
🚨 opinions stated as fact, for brevity sake.

The heuristic should be:

- is this file regularly edited by humans?
  - if yes, include in formatting set
  - if no, it implies it's a machine mantained file (e.g. lockfile), and should not partake in formatting

the formatting and linting separation is valid, and thus these two workstreams should consider these files independently.

**Comment by ematipico** [comment]
>  I didn't think biome would prevent formatting of the package.json, and prettier has no problem doing it.

npm and prettier apply a special formatting to the package.json. If we decided to format the file, this could break that special formatting every time you install a dependency.

**Comment by beeb** [comment]
@cdaringe this file gets edited to add script snippets on a somewhat regular basis in my projects.

@ematipico wouls there be a way to format this in a way that is compatible with npm? If not, then I get your point, the file changes when dependencies are added through the cli.

Final thought : why does the biome extension not consider this file as being on the ignore list then?

**Comment by ematipico** [comment]
> Final thought : why does the biome extension not consider this file as being on the ignore list then?

Because ignored files are configured by the user, while these files are ignored by Biome. We thought to give different semantics for clarity

**Comment by beeb** [comment]
> Because ignored files are configured by the user, while these files are ignored by Biome. We thought to give different semantics for clarity

Sorry my bad, let me rephrase. If Biome ignores this file automatically, why does the Biome VSCode extension format this file when `Format on save` is enabled?

**Comment by nhedger** [comment]
> > Because ignored files are configured by the user, while these files are ignored by Biome. We thought to give different semantics for clarity
> 
> Sorry my bad, let me rephrase. If Biome ignores this file automatically, why does the Biome VSCode extension format this file when `Format on save` is enabled?

This could be a bug. We have an [open issue](https://github.com/biomejs/biome/issues/1654) where ignored files are being formatted when using the VS Code extension.

**Comment by Conaclos** [comment]
NPM now respects the actual indentation of `package.json`. I can't remember when this was fixed, I think it was several years ago.
I don't think we need to do any special formatting or protect the file anymore.

I know we had users requesting to ignore files such as `package.json`. However, we also have users requesting the opposite. While a user can ignore a file using `files.ignore` or `formatter.ignore`, a user cannot format a protected file.

I see several options here:

1. Remove `package.json` from the protected file set
2. Add `files.ignoreProtected` in the config file
3. Get rid of the concept of protected files
4. Same that (2) and introduce a default Biome config that a user can `extends` from
    This default config could ignore the actual protected files or apply special formatting to files such as `package.json` using the `overrides`.

**Comment by ematipico** [comment]
> Get rid of the concept of protected files

I am up for it, but I think this requires a separate discussion. We have other files, `tsconfig.json` (a JSON file parsed as JSONC file), and more. 

> Remove package.json from the protected file set

Let's do it. If anyone wants to contribute here, this should be very easy: 

https://github.com/biomejs/biome/blob/9aece50f85160b7073e07ce49560608a9b62b6bd/crates/biome_service/src/workspace.rs#L112

**Comment by nstepien** [comment]
FWIW biome and prettier will format `package.json` differently.

For example with biome:
```json
{
  "sideEffects": ["./src/**/*.css"]
}
```
And with prettier:
```json
{
  "sideEffects": [
    "./src/**/*.css"
  ]
}
```

I could be wrong but I believe prettier simply does `JSON.stringify(package, null, 2)`.

**Comment by ematipico** [comment]
Which is what npm does and why I was hesitant about it.

**Comment by beeb** [comment]
Like discussed, anyone who doesn't want their package.json formatted can add it to the ignore list. People have a choice now :)

**Comment by Conaclos** [comment]
I missed this issue. Another approach is to keep `package.json` as protected and adding an option `ignoreProtected`.
